packages/shared/src/hooks/useChecklist.ts

```diff
@@ -45,8 +45,20 @@ const useChecklist = ({ steps }: UseChecklistProps): UseChecklist => {
   }, [steps]);
 
   return useMemo(() => {
+    const sortedStepsByCompletion = [...steps].sort((a, b) => {
+      if (a.action.completedAt && !b.action.completedAt) {
+        return -1;
+      }
+
+      if (!a.action.completedAt && b.action.completedAt) {
+        return 1;
+      }
+
+      return 0;
+    });
```

Comment:
Wouldn't this already work?

```
steps.sort((a, b) => a.completedAt - b.completedAt);
```

Reply:
I think it would but personally I like verbosity when not working with numbers.

As for API, I think doing it here just takes care of it from single point in code as said in description of this PR, we would also have to do sorting logic in optimistic mutation in completeAction.

> I evaluated to do this on the API side but since user actions on the API are generic interface and they just happen to be sorted in checklist steps based on completion I decided to do it inside useChecklist hook that contains other checklist specific logic